backends/tfhe-cuda-backend/build.rs

```diff
@@ -60,31 +60,21 @@ fn main() {
 
         let out_path = PathBuf::from("src").join("bindings.rs");
 
-        // Check modification times
-        let header_modified = fs::metadata(header_path).unwrap().modified().unwrap();
-        let bindings_modified = if out_path.exists() {
-            fs::metadata(&out_path).unwrap().modified().unwrap()
-        } else {
-            std::time::SystemTime::UNIX_EPOCH // If bindings file doesn't exist, consider it older
-        };
-        // Regenerate bindings only if header has been modified
-        if header_modified > bindings_modified {
-            let bindings = bindgen::Builder::default()
-                .header(header_path)
-                .clang_arg("-x")
-                .clang_arg("c++")
-                .clang_arg("-std=c++17")
-                .clang_arg("-I/usr/include")
-                .clang_arg("-I/usr/local/include")
-                .ctypes_prefix("ffi")
-                .raw_line("use crate::ffi;")
-                .generate()
-                .expect("Unable to generate bindings");
+        let bindings = bindgen::Builder::default()
+            .header(header_path)
+            .clang_arg("-x")
+            .clang_arg("c++")
+            .clang_arg("-std=c++17")
+            .clang_arg("-I/usr/include")
+            .clang_arg("-I/usr/local/include")
+            .ctypes_prefix("ffi")
+            .raw_line("use crate::ffi;")
+            .generate()
+            .expect("Unable to generate bindings");
```

Comment:
This will generate the bindings every compilation? wouldn't that affect compiling times?

Reply:
it only runs if the build script is run, if it needs to run it either means you are building the cuda code (slow) or you are regenerating the binding and I don't think it's that slow, to be checked, it's a proposal to avoid missing binding regen

Reply:
however for whatever reason the git from the pcc checks is failing, but I don't get the issue locally

Reply:
https://github.com/zama-ai/tfhe-rs/actions/runs/11387881080/job/31683686190?pr=1694#step:9:683

Reply:
ah looks like it does not output the same rust file format

Reply:
it's a step before the check that modifies the file, very weird

Reply:
if I run locally make pcc_gpu, I get : Generated bindings have changed! Please run 'git add backends/tfhe-cuda-backend/src/bindings.rs' and commit the changes.

Reply:
what's the diff ?

Reply:
not sure what are all those for

Reply:
unfurtonately for you it's a change linked to the headers on your system and the export of macros by the binding generator